tools/cli/README.md

```diff
@@ -0,0 +1,71 @@
+## What
+Cadence CLI provide an command-line tool for users to perform various tasks on Cadence.  
+Users can perform operations like register, update and describe on domain;  
+also start workflow, show workflow history, signal workflow ... and many other tasks.    
+
+## How
+- Run `make bins`
+- You should see an executable `cadence-cli`
+
+## Quick Start
+Run `./cadence-cli` to view help message. There are some top level commands and global options.   
+Run `./cadence-cli domain` to view help message about operations on domain  
+Run `./cadence-cli workflow` to view help message about operations on workflow  
+
+*Note:* make sure you have cadence server running before using CLI 
+
+### Domain operation examples 
+- Register a new domain named "samples-domain":  
+```
+./cadence-cli --domain samples-domain domain register 
+# OR using short alias  
+./cadence-cli --do samples-domain domain re
+```   
+- View "samples-domain" details:   
+```
+./cadence-cli --domain samples-domain domain describe  
+```
+
+### Workflow operation examples
+
+- Start workflow: (for example [this workflow](https://github.com/samarabbas/cadence-samples/blob/master/cmd/samples/recipes/helloworld/helloworld_workflow.go))
+```
+./cadence-cli --domain samples-domain workflow start --tl helloWorldGroup --wt main.Workflow --et 60 --dt 10 -i '"vancexu"'
+
+# see more options for workflow start
+./cadence-cli --domain samples-domain workflow start -h
```

Comment:
No, `cadence workflow start help` won't work because there is no `help` subsubcommands, but only `--help` or `-h` command options. 

When view help of `cadence workflow` it will show below:
```
NAME:
   cadence workflow - Operate cadence workflow

USAGE:
   cadence workflow command [command options] [arguments...]

COMMANDS:
     show             show workflow history
     start            start a new workflow execution
     run              start a new workflow execution and get workflow progress
     cancel, c        cancel a workflow execution
     signal, s        signal a workflow execution
     terminate, term  terminate a new workflow execution
     list             list open or closed workflow executions
     query            query workflow execution

OPTIONS:
   --help, -h  show help

```
the usage will indicate how the help option can be used